src/Http/Http.Results/src/ProblemHttpResult.cs

```diff
@@ -37,7 +41,7 @@ internal ProblemHttpResult(ProblemDetails problemDetails)
     /// <summary>
     /// Gets the HTTP status code.
     /// </summary>
-    public int? StatusCode => ProblemDetails.Status;
+    public int StatusCode => ProblemDetails.Status!.Value;
```

Comment:
We set the default in the constructor, and it will not be null https://github.com/dotnet/aspnetcore/blob/d761daa251ba239ac274fb8709b38242007ed5c0/src/Http/Http.Results/src/HttpResultsHelper.cs#L131

But sure, I can add a check if you prefer.